docs: add a guide for HAR record/replay (#14800)
microsoft/playwright: docs/src/network.md

<br/>

## Record and replay requests

> You can record network activity as an HTTP Archive file (HAR). Later on, this archive can be used to supply mock responses to network requests. You'll need:

Reviewer: ```suggestion
You can record network activity as an HTTP Archive file (HAR). Later on, this archive can be used to mock responses to the network requests. You'll need to:
```

1. Record a HAR file.
1. Commit the HAR file alongside the tests.

> 1. Use the HAR file in the tests.

Reviewer: ```suggestion
1. Route requests using the saved HAR files in the tests.
```

```bash js
# Save API requests from example.com as "example.har" archive.
npx playwright open --save-har=example.har --save-har-glob="**/api/**" https://example.com
```

```bash java
# Save API requests from example.com as "example.har" archive.
mvn exec:java -e -Dexec.mainClass=com.microsoft.playwright.CLI -Dexec.args="open --save-har=example.har --save-har-glob='**/api/**' https://example.com"
```

```bash python
# Save API requests from example.com as "example.har" archive.
playwright open --save-har=example.har --save-har-glob="**/api/**" https://example.coms
```

```bash csharp
# Save API requests from example.com as "example.har" archive.
pwsh bin\Debug\netX\playwright.ps1 open --save-har=example.har --save-har-glob="**/api/**" https://example.com
```

### Recording HAR with a script

Alternatively, instead of using the CLI, you can record HAR programmatically. Pass [`option: har`] option when creating a [BrowserContext] with [`method: Browser.newContext`] to create an archive.

```js
const context = await browser.newContext({

>   recordHar: { path: 'example.har' }

Reviewer: ```suggestion
  recordHar: { path: 'example.har', urlFilter: '**/*.json' }
```
